pkg/collector/harbor_events.go

```diff
@@ -0,0 +1,508 @@
+		Vulnerability: &vulnerability.Details{
+			Severity: vulnerabilitySeverity,
+			PackageIssue: []*vulnerability.PackageIssue{
+				{
+					AffectedLocation: &vulnerability.VulnerabilityLocation{
+						CpeUri:  "TODO",
+						Package: "TODO",
+						Version: &packag.Version{
+							Kind: packag.Version_NORMAL,
+							Name: "TODO",
+						},
+					},
+				},
+			},
+		},
+	}
+	return vulnerabilityDetails
+}
+
+func (t *HarborEventCollector) getVulnerabilitySeverity(v string) vulnerability.Severity {
+	switch v {
+	case HarborSeverityCritical:
+		return vulnerability.Severity_CRITICAL
+	case HarborSeverityHigh:
+		return vulnerability.Severity_HIGH
+	case HarborSeverityMedium:
+		return vulnerability.Severity_MEDIUM
+	case HarborSeverityLow:
+		return vulnerability.Severity_LOW
+	case HarborSeverityNegligible:
+		return vulnerability.Severity_MINIMAL
+	case HarborSeverityNone:
+		return vulnerability.Severity_MINIMAL
+	default:
+		return vulnerability.Severity_SEVERITY_UNSPECIFIED
+	}
+}
+
+// Assumes Harbor admin creds are deployed in the same namespace as the Collector CR
+func (t *HarborEventCollector) getHarborCredentials(ctx context.Context, secretname string, namespace string) (string, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return "", err
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return "", err
+	}
+	secrets, err := clientset.CoreV1().Secrets(namespace).Get(secretname, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	return string(secrets.Data["HARBOR_ADMIN_PASSWORD"]), nil
+}
+
+func (t *HarborEventCollector) getHostName(ingressname string, namespace string) (string, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return "", err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return "", err
+	}
+
+	ingress, err := clientset.ExtensionsV1beta1().Ingresses(namespace).Get(ingressname, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	return "https://" + ingress.Spec.Rules[0].Host, nil
+}
+
+func (t *HarborEventCollector) getProjectID(name string, url string) (string, error) {
+	client := &http.Client{}
+	var projects []Project
+	var projectID string
+
+	req, err := http.NewRequest("GET", url+"/api/projects/", nil)
+	if err != nil {
+		return "", err
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	projectList, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	err = json.Unmarshal(projectList, &projects)
+	if err != nil {
+		return "", err
+	}
+
+	for _, p := range projects {
+		if p.Name == name {
+			projectID = strconv.Itoa(p.ProjectID)
+		}
+	}
+	return projectID, nil
+}
+
+func (t *HarborEventCollector) getWebhookPolicyID(projectID string, url string, harborCreds string) (string, error) {
+	client := &http.Client{}
+	webhookPolicyIDURL := url + "/api/projects/" + projectID + "/webhook/policies"
```

Comment:
when building this url (this time and all other times), can you use `fmt.Sprintf` for it?